todo-app/src/Components/ToDoList.tsx

```diff
@@ -1,96 +1,86 @@
-import { useEffect, useState } from "react";
 import ToDo from "./ToDo";
-import IToDo from "../interfaces/Todo";
 import useFetch from "../hooks/useFetch";
+import axios from "axios";
+import { useNavigate } from "react-router-dom";
 
 const ToDoList = () => {
-  const [todo, setToDo] = useState("");
-  const [todolist, setToDoList] = useState<IToDo[]>([]);
   const url = "http://localhost:8000/todo";
-  const { response, error, loader } = useFetch(url);
+  const { response, error, loader, refetch } = useFetch(url);
+  const navigate = useNavigate();
 
-  useEffect(() => {
-    if (!error) {
-      setToDoList(response);
-    } else {
-      alert(error);
-    }
-  }, [response, error]);
-
-  const handleInput = (event: any) => {
-    setToDo(event.target.value);
+  const deleteToDo = async (key: number) => {
+    await axios
+      .delete(`${url}/${key}`)
+      .then((data) => console.log(data.data))
+      .catch((err) => console.log(err));
+    refetch({});
   };
 
-  const saveInput = () => {
-    if (todo) {
-      const curr = {
-        text: todo,
-        isCompleted: false,
-      };
-      setToDoList([...todolist, curr]);
-      setToDo("");
+  const handleCheck = (key: number) => {
+    const currentTodo = response.find((item) => item.id === key);
+    if (currentTodo) {
+      currentTodo.isCompleted = !currentTodo?.isCompleted;
+      const currid = currentTodo.id;
+      axios
+        .put(`${url}/${currid}`, currentTodo)
+        .then(() => refetch({}))
+        .catch((err) => alert(err));
+      console.log(currentTodo);
     } else {
-      alert("enter the todo you want to add !");
+      alert("not found");
     }
   };
 
-  const deleteToDo = (index: any) => {
-    const updatedTodos = [...todolist];
-    updatedTodos.splice(index, 1);
-    setToDoList(updatedTodos);
-  };
-
-  const handleCheck = (index: any) => {
-    const currentTodo = todolist[index];
-    currentTodo.isCompleted = !currentTodo.isCompleted;
-    setToDoList([...todolist]);
-  };
-
   return (
     <>
       <div>
-        <input type="text" value={todo} onChange={handleInput} />
-        <button type="button" onClick={saveInput}>
-          Add
-        </button>
-        {loader ? (
-          <h3>loading....</h3>
+        <button onClick={() => navigate("/create")}>Add</button>
+      </div>
+      <div>
+        {!error ? (
```

Comment:
{ error  &&  <> error code </> }
{loader && <h3>loading....</h3> }
{ !error && !loader && () }

Reply:
Have improved the code but not exactly as you told but it is more readable as you told